src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -451,13 +461,18 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
         logger.print("\n[JaCoCo plugin] Saving matched class directories for class-pattern: " + classPattern + ": ");
         for (FilePath file : matchedClassDirs) {
             dir.saveClassesFrom(file);
-            logger.print(" " + file);
```

Comment:
these should only be the top-level class dirs, e.g. build/classes, not each separate directory underneath. This sounds like you configured it in a way so that it selected all those separately one-by-one instead of only the top-level dir. Can you try to configure it differently and see if the printout looks sane then while still collecting coverage?